### ISSUE TITLE
CMS pages: add options to show when created and last edited and by whom, with editable defaults

### ISSUE THREAD

**Issue by CrispinF** [cloudscribe/cloudscribe.SimpleContent#487]
Four options in Settings:
Show created by?
Show created date?
Show last modified by?
Show last modified date?
To be rendered below content.
There should be settable default values for these under Content Settings, as these tend to be used or not used according to some editorial policy. But this default could be overridden on particular pages.

**Comment by markidox** [comment]
[I've created a pull request for this](https://github.com/cloudscribe/cloudscribe.SimpleContent/pull/601). It will need quite a bit of testing as it touches various parts of the Simple Content system.
The settings (dubbed 'Editorial Settings') can be set from the Simple Content settings within admin, this should set them across standard pages and templates.
The settings also appear on each page, if they haven't been altered for that page, they will show the default settings.
New pages should be created, both from templates and from 'standard'.
Both sets of pages should be edited and deleted.
The Editorial Settings should be set/unset at a settings and individual page level.

**Comment by JimKerslake** [comment]
No matter what I do I can't persuade the migration here to run (MSSQL) though there is no error logged.  Can't really go ahead with merging this at the moment.

I thought it was perhaps because other migrations have been declared in simplecontent with later dates on them.... 
but it's not supposed to work that way... at least I thought not.

**Comment by JimKerslake** [comment]
I did try to take the out-of-sequence migrations and re-date their class names so they would be the latest ones - but still something (maybe the modelsnapshot file) was not going to allow them to run.  I guess sequential date order (with no backfilling) must apply to migrations within any assembly (but not across assemblies?) though the MigrationHistory table is pretty useless in helping you to diagnose that, since it has just the one meaningful field.

I would actually make a fresh git branch off master, re-make the dbContext changes and migrations, and manually copy across the other code changes then try again...  tedious.

Perhaps we need to git-merge PRs in the date order of any migrations they may contain  :/

**Comment by JimKerslake** [comment]
When naming migrations we should perhaps adopt the policy of including the project name:
20190312154938_simplecontent-20190312
because that helps a lot when staring at the MigrationHistory table.

**Comment by JimKerslake** [comment]
****  See:  See  https://github.com/cloudscribe/cloudscribe.SimpleContent/pull/601#issuecomment-2452099617

**Comment by JimKerslake** [comment]
Nope migrations still not working for me... the PR still has them wrongly as booleans?
https://github.com/cloudscribe/cloudscribe.SimpleContent/pull/601#issuecomment-2452099617

**Comment by JimKerslake** [comment]
I have a feeling that this is going to require more testing of actual behaviour - i.e. when these things do and don't appear.

What is the point of having these settings site-wide if individual pages can always override them ?   
i.e. if an individual page's settings always override the site-wide settings to the positive **OR** to the negative, then...
that individual page's settings will always win out over the site-wide ones whichever way they go, so...
site-wide ones never get used?

Or is it only ever additive / positive?

**Comment by markidox** [comment]
I'm not 100% sure what you're asking but it should work that you set them site-wide (to be on for example) and then if you decide you don't want them to display on a particular page, you can turn them off for that particular page, or vice-versa I guess, but I think the use case would more likely be you want to show some/all of the items for the majority of pages with some pages (perhaps that don't get updated that often) don't show those items.

**Comment by JimKerslake** [comment]
Yes it's a bit confusing - I may have a misunderstanding of  @CrispinF 's  specification here... 

I can just envisage a slightly odd competition between these two places for setting the same thing, if we reckon we can override to the positive or to the negative. 

Any individual page's editing UI does by default seem to show the 4 checkboxes checked according to the preference from the site-wide settings - yep - that makes sense.

But then if you make any content change at all to the page and post it again (regardless of whether you care about those checkboxes) then the 4 booleans for the individual page get explicitly populated in cs_page - i,e, this page is in effect now overriding the preference of the site-wide content settings, and I guess there's no way back from that..?

And indeed if I make a new page, then this will inherit the site-wide settings, but that seems to populate the 4 individual bools at page level - i,e, my new page will always be overriding whatever is set at site level...? 

That may all be fine... or I may be misinterpreting... 
really needs a bit of spec-clarification perhaps?

**Comment by markidox** [comment]
Ah ok ,so there was a reason behind that and, off the top of my head, it was because in order to enable overriding of those elements on an individual page those elements needed to be against each page.
If you change anything site wide I *think* (can't quite remember as it was a bit ago) it checks to see if you have changed any page elements and, if not, it changes them according to the changes you are making site-wide, if you have changed them it leaves them alone. I'd have to dig in to what I did to clarify that, or fire it up and run it but I'm just in the middle of this wysiwyg editor stuff at the min.

**Comment by CrispinF** [comment]
@markidox looking good, however as discussed, please could you relocate the page-specific settings for this to the Settings tab, not the Content tab. I'd suggest inserting the new settings between these two:
![image](https://github.com/user-attachments/assets/163bcb59-4571-4b5b-8c4c-40569b9a3166)

And the display "By" fields seem to be using the user's UserName property, whereas they should be using the DisplayName.

**Comment by markidox** [comment]
Have moved the items into the settings page between the two options you suggested, PR: https://github.com/exeGesIS-SDM/hauc/pull/486 @JimKerslake 

With regards to the "By" fields, this data is coming from the page fields rather than the user data. So the page tells me who has updated and who created and it's using the user name property. If I had to guess why it would be because you could have display names that are the same but usernames can't be?

**Comment by JimKerslake** [comment]
Deployed outstanding PR from branch 497-4 to develop / cs-current testing sites
